arwen/hosttest/execution_gas_test.go

```diff
@@ -389,6 +391,148 @@ func TestGasUsed_AsyncCall(t *testing.T) {
 		})
 }
 
+func TestGasUsed_AsyncCall_CrossShard(t *testing.T) {
```

Comment:
I initially said that the three executions should stay together in the same test, but it might be better to have them as separate tests, after all.